models/etablissement.ts

```diff
@@ -24,31 +33,107 @@ const getEtablissementFromSlug = async (
   const siret = verifySiret(slug);
 
   const isBot = options?.isBot || false;
-  const shouldNotUseInsee = process.env.INSEE_ENABLED === 'disabled';
 
-  const etablissement =
-    shouldNotUseInsee || isBot
-      ? await getEtablissementRechercheEntreprise(siret)
-      : await getEtablissement(siret);
+  const etablissement = fetchFromClients(siret, isBot);
 
   return etablissement;
 };
 
 /**
  * Return an Etablissement for a given siret
  */
-const getEtablissement = async (siret: Siret): Promise<IEtablissement> => {
+const fetchFromClients = async (
```

Comment:
It seems to me that the logic for orchestrating the various calls is exactly the same as for `unite-legale.ts`. Could we factor this into a function? Something like : 

```typescript
function fetchFromInseeOrchestrator<IRecherche | IInsee>(
   isBot: boolean, 
   apiRecherche: () => Promise<IRecherche | IAPINotResponding>,
   insee (useFallback: boolean) => Promise<IInsee | IAPINotResponding>,
) : Promise<IRecherche | IInsee | IAPINotResponding>